src/gui/guiFormSpecMenu.cpp

```diff
@@ -766,108 +766,75 @@ void GUIFormSpecMenu::parseScrollBarOptions(parserData* data, const std::string
 
 void GUIFormSpecMenu::parseImage(parserData* data, const std::string &element)
 {
-	std::vector<std::string> parts = split(element,';');
-
-	if ((parts.size() == 3) ||
-		((parts.size() > 3) && (m_formspec_version > FORMSPEC_API_VERSION)))
-	{
-		std::vector<std::string> v_pos = split(parts[0],',');
-		std::vector<std::string> v_geom = split(parts[1],',');
-		std::string name = unescape_string(parts[2]);
+	std::vector<std::string> parts = split(element, ';');
 
-		MY_CHECKPOS("image", 0);
-		MY_CHECKGEOM("image", 1);
+	if (parts.size() < 3 || parts.size() > 4 ||
```

Comment:
This is not correct.

Fix:

```c++
if ((parts.size() < 3 || parts.size() > 4) && (m_formspec_version <= FORMSPEC_API_VERSION || parts.size() < 3)) {
```

Also, this should really be a macro or something. I like how my FormspecElement does it

Reply:
Yes, the FormspecElement was very nice.  I made something with `std::initializer_list` in my partial refactor like `checkSize(m_formspec_version, {3, 4})`, but that might count as too C++y.

Reply:
Actually, I think the logic in this is clearer despite being longer:


```c++
if (!((parts.size() >= 3 && parts.size() <= 4) || (m_formspec_version > FORMSPEC_API_VERSION && parts.size() > 4))) {
```